### ISSUE TITLE
Failed to solve conda environment for mcc_relocate2

### ISSUE THREAD

**Issue by s-andrews** [bergmanlab/mcclintock#113]
I've installed mcclintock and am trying to run the test suite but am getting a failure from trying to build the mcc_relocate2 conda environment.

```
[andrewss@headstone mcclintock]$ conda activate mcclintock
(mcclintock) [andrewss@headstone mcclintock]$ python3 mcclintock.py \
>     -r test/sacCer2.fasta \
>     -c test/sac_cer_TE_seqs.fasta \
>     -g test/reference_TE_locations.gff \
>     -t test/sac_cer_te_families.tsv \
>     -1 test/SRR800842_1.fastq.gz \
>     -2 test/SRR800842_2.fastq.gz \
>     -p 4 \
>     -o ../test_results/
SETUP            checking fasta: /bi/group/bioinf/Jon_Houseley/McClintock/mcclintock/test/sacCer2.fasta
SETUP            checking fastq: /bi/group/bioinf/Jon_Houseley/McClintock/mcclintock/test/SRR800842_1.fastq.gz
SETUP            checking fastq: /bi/group/bioinf/Jon_Houseley/McClintock/mcclintock/test/SRR800842_2.fastq.gz
SETUP            checking fasta: /bi/group/bioinf/Jon_Houseley/McClintock/mcclintock/test/sac_cer_TE_seqs.fasta
SETUP            checking locations gff: /bi/group/bioinf/Jon_Houseley/McClintock/mcclintock/test/reference_TE_locations.gff
SETUP            checking taxonomy TSV: /bi/group/bioinf/Jon_Houseley/McClintock/mcclintock/test/sac_cer_te_families.tsv
SETUP            McClintock Version: 3ffe2f36cf32e846caab417307a301abd784a9b0
CreateCondaEnvironmentException:
Could not create conda environment from /bi/group/bioinf/Jon_Houseley/McClintock/mcclintock/install/envs/mcc_relocate2.yml:
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working...
Building graph of deps:   0%|          | 0/13 [00:00<?, ?it/s]
Examining @/linux-64::__archspec==1=x86_64:   0%|          | 0/13 [00:00<?, ?it/s]
Examining bedtools==2.26.0-0:   8%|▊         | 1/13 [00:00<00:00, 5683.34it/s]
Examining perl[version='>=5.26.2,<5.26.3.0a0']:  15%|█▌        | 2/13 [00:00<00:00, 1219.45it/s]
Examining blat==35:  23%|██▎       | 3/13 [00:00<00:00, 56.19it/s]
Examining pysam==0.9.0:  31%|███       | 4/13 [00:00<00:00, 67.75it/s]
Examining samtools==1.3-0:  38%|███▊      | 5/13 [00:00<00:00, 20.45it/s]
Examining samtools==1.3-0:  46%|████▌     | 6/13 [00:00<00:00, 24.53it/s]
Examining @/linux-64::__glibc==2.17=0:  46%|████▌     | 6/13 [00:00<00:00, 24.53it/s]
Examining seqtk==1.2-0:  54%|█████▍    | 7/13 [00:00<00:00, 24.53it/s]
Examining @/linux-64::__unix==0=0:  62%|██████▏   | 8/13 [00:00<00:00, 24.53it/s]
Examining bowtie2==2.2.8:  69%|██████▉   | 9/13 [00:00<00:00, 24.53it/s]
Examining @/linux-64::__linux==3.10.0=0:  77%|███████▋  | 10/13 [00:00<00:00, 24.53it/s]
Examining @/linux-64::__linux==3.10.0=0:  85%|████████▍ | 11/13 [00:00<00:00, 30.06it/s]
Examining python==2.7.12:  85%|████████▍ | 11/13 [00:00<00:00, 30.06it/s]
Examining bwa==0.6.2-0:  92%|█████████▏| 12/13 [00:00<00:00, 30.06it/s]


Determining conflicts:   0%|          | 0/13 [00:00<?, ?it/s]
Examining conflict for bedtools samtools pysam blat seqtk python bwa:   0%|          | 0/13 [00:00<?, ?it/s]
Examining conflict for bowtie2 bedtools pysam bwa:   8%|▊         | 1/13 [00:00<00:01,  6.10it/s]
Examining conflict for bowtie2 bedtools pysam bwa:  15%|█▌        | 2/13 [00:00<00:00, 12.18it/s]
Examining conflict for bedtools pysam perl blat samtools seqtk bowtie2 python bwa:  15%|█▌        | 2/13 [00:00<00:00, 12.18it/s]
Examining conflict for bedtools samtools pysam bowtie2 bwa:  23%|██▎       | 3/13 [00:00<00:00, 12.18it/s]
Examining conflict for bedtools samtools pysam bowtie2 bwa:  31%|███       | 4/13 [00:00<00:01,  6.91it/s]
Examining conflict for bedtools samtools pysam blat seqtk bowtie2 python bwa:  31%|███       | 4/13 [00:00<00:01,  6.91it/s]
Examining conflict for bedtools samtools pysam blat seqtk bowtie2 python bwa:  38%|███▊      | 5/13 [00:00<00:01,  6.52it/s]
Examining conflict for pysam bowtie2 perl:  38%|███▊      | 5/13 [00:00<00:01,  6.52it/s]
Examining conflict for pysam bowtie2 perl:  46%|████▌     | 6/13 [00:00<00:01,  6.21it/s]
Examining conflict for samtools perl bwa:  46%|████▌     | 6/13 [00:00<00:01,  6.21it/s]
Examining conflict for samtools perl bwa:  54%|█████▍    | 7/13 [00:00<00:00,  6.99it/s]
Examining conflict for samtools perl bwa __glibc:  54%|█████▍    | 7/13 [00:01<00:00,  6.99it/s]
Examining conflict for samtools perl bwa __glibc:  62%|██████▏   | 8/13 [00:01<00:00,  6.99it/s]
Examining conflict for bowtie2 python pysam:  62%|██████▏   | 8/13 [00:01<00:00,  6.99it/s]
Examining conflict for samtools pysam:  69%|██████▉   | 9/13 [00:01<00:00,  6.99it/s]
Examining conflict for samtools pysam:  77%|███████▋  | 10/13 [00:01<00:00,  8.51it/s]
Examining conflict for samtools bowtie2 pysam python:  77%|███████▋  | 10/13 [00:01<00:00,  8.51it/s]
Examining conflict for bowtie2 pysam:  85%|████████▍ | 11/13 [00:01<00:00,  8.51it/s]
Examining conflict for bowtie2 pysam:  92%|█████████▏| 12/13 [00:01<00:00,  7.42it/s]

Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed
Solving environment: ...working...

Building graph of deps:   0%|          | 0/13 [00:00<?, ?it/s]
Examining @/linux-64::__archspec==1=x86_64:   0%|          | 0/13 [00:00<?, ?it/s]
Examining bedtools==2.26.0-0:   8%|▊         | 1/13 [00:00<00:00, 6955.73it/s]
Examining perl[version='>=5.26.2,<5.26.3.0a0']:  15%|█▌        | 2/13 [00:00<00:00, 1558.35it/s]
Examining blat==35:  23%|██▎       | 3/13 [00:00<00:00, 56.66it/s]
Examining pysam==0.9.0:  31%|███       | 4/13 [00:00<00:00, 68.13it/s]
Examining samtools==1.3-0:  38%|███▊      | 5/13 [00:00<00:00, 20.58it/s]
Examining samtools==1.3-0:  46%|████▌     | 6/13 [00:00<00:00, 24.69it/s]
Examining @/linux-64::__glibc==2.17=0:  46%|████▌     | 6/13 [00:00<00:00, 24.69it/s]
Examining seqtk==1.2-0:  54%|█████▍    | 7/13 [00:00<00:00, 24.69it/s]
Examining @/linux-64::__unix==0=0:  62%|██████▏   | 8/13 [00:00<00:00, 24.69it/s]
Examining bowtie2==2.2.8:  69%|██████▉   | 9/13 [00:00<00:00, 24.69it/s]
Examining @/linux-64::__linux==3.10.0=0:  77%|███████▋  | 10/13 [00:00<00:00, 24.69it/s]
Examining @/linux-64::__linux==3.10.0=0:  85%|████████▍ | 11/13 [00:00<00:00, 30.07it/s]
Examining python==2.7.12:  85%|████████▍ | 11/13 [00:00<00:00, 30.07it/s]
Examining bwa==0.6.2-0:  92%|█████████▏| 12/13 [00:00<00:00, 30.07it/s]


Determining conflicts:   0%|          | 0/13 [00:00<?, ?it/s]
Examining conflict for bedtools samtools pysam blat seqtk python bwa:   0%|          | 0/13 [00:00<?, ?it/s]
Examining conflict for bowtie2 bedtools pysam bwa:   8%|▊         | 1/13 [00:00<00:02,  5.90it/s]
Examining conflict for bowtie2 bedtools pysam bwa:  15%|█▌        | 2/13 [00:00<00:00, 11.79it/s]
Examining conflict for bedtools pysam perl blat samtools seqtk bowtie2 python bwa:  15%|█▌        | 2/13 [00:00<00:00, 11.79it/s]
Examining conflict for bedtools samtools pysam bowtie2 bwa:  23%|██▎       | 3/13 [00:00<00:00, 11.79it/s]
Examining conflict for bedtools samtools pysam bowtie2 bwa:  31%|███       | 4/13 [00:00<00:01,  6.73it/s]
Examining conflict for bedtools samtools pysam blat seqtk bowtie2 python bwa:  31%|███       | 4/13 [00:00<00:01,  6.73it/s]
Examining conflict for bedtools samtools pysam blat seqtk bowtie2 python bwa:  38%|███▊      | 5/13 [00:00<00:01,  6.33it/s]
Examining conflict for pysam bowtie2 perl:  38%|███▊      | 5/13 [00:00<00:01,  6.33it/s]
Examining conflict for pysam bowtie2 perl:  46%|████▌     | 6/13 [00:00<00:01,  6.01it/s]
Examining conflict for samtools perl bwa:  46%|████▌     | 6/13 [00:01<00:01,  6.01it/s]
Examining conflict for samtools perl bwa:  54%|█████▍    | 7/13 [00:01<00:00,  6.75it/s]
Examining conflict for samtools perl bwa __glibc:  54%|█████▍    | 7/13 [00:01<00:00,  6.75it/s]
Examining conflict for samtools perl bwa __glibc:  62%|██████▏   | 8/13 [00:01<00:00,  6.76it/s]
Examining conflict for bowtie2 python pysam:  62%|██████▏   | 8/13 [00:01<00:00,  6.76it/s]
Examining conflict for samtools pysam:  69%|██████▉   | 9/13 [00:01<00:00,  6.76it/s]
Examining conflict for samtools pysam:  77%|███████▋  | 10/13 [00:01<00:00,  8.26it/s]
Examining conflict for samtools bowtie2 pysam python:  77%|███████▋  | 10/13 [00:01<00:00,  8.26it/s]
Examining conflict for bowtie2 pysam:  85%|████████▍ | 11/13 [00:01<00:00,  8.26it/s]
Examining conflict for bowtie2 pysam:  92%|█████████▏| 12/13 [00:01<00:00,  7.23it/s]

Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package isl conflicts for:
bedtools==2.26.0-0 -> libgcc -> isl
bowtie2==2.2.8 -> libgcc -> isl
bwa==0.6.2-0 -> libgcc -> isl
pysam==0.9.0 -> libgcc -> isl

Package zlib conflicts for:
python==2.7.12 -> zlib=1.2
blat==35 -> libpng -> zlib[version='1.2.*|1.2.11|>=1.2.11,<1.3.0a0|>=1.2.12,<1.3.0a0|1.2.8|>=1.2.13,<1.3.0a0']
bowtie2==2.2.8 -> python=3.5 -> zlib[version='1.2.*|1.2.11|>=1.2.11,<1.3.0a0|>=1.2.13,<1.3.0a0|1.2.11.*|1.2.8']
blat==35 -> zlib=1.2.11
bedtools==2.26.0-0 -> zlib
pysam==0.9.0 -> bcftools==1.3 -> zlib[version='1.2.*|1.2.11|1.2.11.*|>=1.2.11,<1.3.0a0|1.2.8|>=1.2.13,<1.3.0a0|>=1.2.12,<1.3.0a0|1.2.8.*']
bwa==0.6.2-0 -> zlib[version='1.2.8.*|>=1.2.11,<1.3.0a0']
seqtk==1.2-0 -> zlib=1.2.11
samtools==1.3-0 -> zlib[version='>=1.2.11,<1.3.0a0']

Package _libgcc_mutex conflicts for:
samtools==1.3-0 -> libgcc-ng[version='>=10.3.0'] -> _libgcc_mutex[version='*|0.1',build='main|conda_forge|main']
bwa==0.6.2-0 -> libgcc-ng[version='>=4.9'] -> _libgcc_mutex[version='*|0.1',build='main|conda_forge|main']
perl[version='>=5.26.2,<5.26.3.0a0'] -> libgcc-ng[version='>=7.2.0'] -> _libgcc_mutex[version='*|0.1',build='main|conda_forge|main']

Package libzlib conflicts for:
blat==35 -> libpng -> libzlib[version='1.2.11|1.2.11|1.2.11|>=1.2.12,<1.3.0a0|>=1.2.13,<1.3.0a0|1.2.13|1.2.12|1.2.12|1.2.12|1.2.12|1.2.12',build='h166bdaf_0|h166bdaf_1|h166bdaf_2|h166bdaf_3|h166bdaf_4|h36c2ea0_1012|h36c2ea0_1013|h166bdaf_1014|h166bdaf_4']
bedtools==2.26.0-0 -> zlib -> libzlib[version='1.2.11|1.2.11|1.2.11|1.2.12|1.2.12|1.2.12|1.2.12|1.2.12|1.2.13',build='h36c2ea0_1012|h36c2ea0_1013|h166bdaf_1014|h166bdaf_1|h166bdaf_3|h166bdaf_4|h166bdaf_2|h166bdaf_0']
seqtk==1.2-0 -> zlib=1.2.11 -> libzlib[version='1.2.11|1.2.13|1.2.12|1.2.12|1.2.12|1.2.12|1.2.12',build='h166bdaf_0|h166bdaf_1|h166bdaf_2|h166bdaf_3|h166bdaf_4|h36c2ea0_1012|h36c2ea0_1013|h166bdaf_1014|h166bdaf_4']
bwa==0.6.2-0 -> zlib[version='>=1.2.11,<1.3.0a0'] -> libzlib[version='1.2.11|1.2.11|1.2.11|1.2.12|1.2.12|1.2.12|1.2.12|1.2.12|1.2.13',build='h36c2ea0_1012|h36c2ea0_1013|h166bdaf_1014|h166bdaf_1|h166bdaf_3|h166bdaf_4|h166bdaf_2|h166bdaf_0']
samtools==1.3-0 -> libzlib[version='>=1.2.11,<1.3.0a0']
python==2.7.12 -> zlib=1.2 -> libzlib[version='1.2.11|1.2.11|1.2.11|1.2.12|1.2.12|1.2.12|1.2.12|1.2.12|1.2.13',build='h36c2ea0_1012|h36c2ea0_1013|h166bdaf_1014|h166bdaf_1|h166bdaf_3|h166bdaf_4|h166bdaf_2|h166bdaf_0']
samtools==1.3-0 -> curl[version='>=7.81.0,<8.0a0'] -> libzlib[version='1.2.11|1.2.11|1.2.11|1.2.12|1.2.12|1.2.12|1.2.12|1.2.12|1.2.13|>=1.2.12,<1.3.0a0|>=1.2.13,<1.3.0a0',build='h36c2ea0_1012|h36c2ea0_1013|h166bdaf_1014|h166bdaf_1|h166bdaf_3|h166bdaf_4|h166bdaf_2|h166bdaf_0']
pysam==0.9.0 -> bcftools==1.3 -> libzlib[version='>=1.2.11,<1.3.0a0|>=1.2.13,<1.3.0a0|>=1.2.12,<1.3.0a0']

Package samtools conflicts for:
pysam==0.9.0 -> samtools==1.3
samtools==1.3-0

Package openssl conflicts for:
samtools==1.3-0 -> curl[version='>=7.81.0,<8.0a0'] -> openssl[version='>=1.1.1n,<1.1.2a|>=1.1.1o,<1.1.2a|>=1.1.1q,<1.1.2a|>=1.1.1s,<1.1.2a|>=3.0.8,<4.0a0|>=3.1.0,<4.0a0|>=3.0.7,<4.0a0|>=3.0.5,<4.0a0|>=3.0.3,<4.0a0|>=3.0.2,<4.0a0|>=3.0.0,<4.0a0']
samtools==1.3-0 -> openssl[version='>=1.1.0,<=1.1.1|>=1.1.1l,<1.1.2a']

Package libffi conflicts for:
pysam==0.9.0 -> python=3.5 -> libffi[version='3.2.*|>=3.2.1,<4.0a0|>=3.3,<3.4.0a0|>=3.2.1,<3.3.0a0|>=3.2.1,<3.3a0|>=3.4,<3.5|>=3.4,<4.0a0']
bowtie2==2.2.8 -> python=3.5 -> libffi[version='3.2.*|>=3.2.1,<4.0a0|>=3.3,<3.4.0a0|>=3.2.1,<3.3.0a0|>=3.2.1,<3.3a0|>=3.4,<3.5|>=3.4,<4.0a0']

Package libgcc-ng conflicts for:
bwa==0.6.2-0 -> libgcc-ng[version='>=4.9']
bwa==0.6.2-0 -> zlib[version='>=1.2.11,<1.3.0a0'] -> libgcc-ng[version='7.2.0.*|>=11.2.0|>=7.5.0|>=7.3.0|>=7.2.0|>=12|>=10.3.0']
samtools==1.3-0 -> libgcc-ng[version='>=10.3.0|>=9.3.0|>=7.5.0|>=7.3.0']
pysam==0.9.0 -> bcftools==1.3 -> libgcc-ng[version='7.2.0.*|>=10.3.0|>=9.3.0|>=7.3.0|>=4.9|>=7.2.0|>=7.5.0|>=11.2.0|>=12|>=9.4.0']
bowtie2==2.2.8 -> libgcc -> libgcc-ng[version='7.2.0.*|>=7.2.0|>=7.5.0|>=7.3.0|>=4.9|>=11.2.0']
python==2.7.12 -> openssl=1.0 -> libgcc-ng[version='7.2.0.*|>=7.2.0|>=7.3.0|>=4.9|>=11.2.0|>=7.5.0|>=12|>=10.3.0']
blat==35 -> libpng -> libgcc-ng[version='7.2.0.*|>=7.2.0|>=7.3.0|>=12|>=7.5.0|>=4.9|>=11.2.0|>=10.3.0']
perl[version='>=5.26.2,<5.26.3.0a0'] -> libgcc-ng[version='>=4.9|>=7.2.0|>=7.5.0|>=7.3.0']
bedtools==2.26.0-0 -> libgcc -> libgcc-ng[version='7.2.0.*|>=11.2.0|>=7.2.0|>=7.5.0|>=7.3.0|>=12|>=10.3.0|>=4.9']
samtools==1.3-0 -> curl[version='>=7.81.0,<8.0a0'] -> libgcc-ng[version='7.2.0.*|>=11.2.0|>=12|>=9.4.0|>=7.2.0|>=4.9']
seqtk==1.2-0 -> zlib=1.2.11 -> libgcc-ng[version='7.2.0.*|>=7.3.0|>=7.5.0|>=7.2.0|>=10.3.0|>=4.9|>=11.2.0|>=12']

Package libstdcxx-ng conflicts for:
samtools==1.3-0 -> ncurses -> libstdcxx-ng[version='7.2.0.*|>=7.2.0|>=7.3.0|>=7.5.0|>=4.9']
bedtools==2.26.0-0 -> libgcc -> libstdcxx-ng[version='>=7.2.0']
bwa==0.6.2-0 -> libgcc -> libstdcxx-ng[version='>=7.2.0']
bowtie2==2.2.8 -> libgcc -> libstdcxx-ng[version='7.2.0.*|>=7.2.0|>=7.5.0|>=7.3.0|>=4.9']
pysam==0.9.0 -> libgcc -> libstdcxx-ng[version='7.2.0.*|>=7.2.0|>=7.5.0|>=7.3.0|>=4.9']

Package perl conflicts for:
perl[version='>=5.26.2,<5.26.3.0a0']
pysam==0.9.0 -> bcftools -> perl
bowtie2==2.2.8 -> perl-threaded -> perl[version='5.22.0.*|>=5.22.0,<5.23.0']

Package ncurses conflicts for:
bowtie2==2.2.8 -> python=3.5 -> ncurses[version='5.9.*|5.9|6.0.*|>=6.0,<7.0a0|>=6.1,<7.0a0|>=6.2,<7.0a0|>=6.1,<7.0.0a0|>=6.3,<7.0a0']
python==2.7.12 -> ncurses=5.9
pysam==0.9.0 -> python=3.5 -> ncurses[version='5.9.*|5.9|6.0.*|>=6.0,<7.0a0|>=6.1,<7.0a0|>=6.2,<7.0a0|>=6.1,<7.0.0a0|>=6.3,<7.0a0|>=6.2,<6.3.0a0|>=6.1,<6.2.0a0|>=5.9,<5.10.0a0']
samtools==1.3-0 -> ncursesThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.17=0
  - feature:|@/linux-64::__glibc==2.17=0
  - bwa==0.6.2-0 -> libgcc-ng[version='>=4.9'] -> __glibc[version='>=2.17']
  - perl[version='>=5.26.2,<5.26.3.0a0'] -> libgcc-ng[version='>=7.2.0'] -> __glibc[version='>=2.17']
  - samtools==1.3-0 -> libgcc-ng[version='>=10.3.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.17



  File "/bi/home/andrewss/mambaforge/envs/mcclintock/lib/python3.8/site-packages/snakemake/deployment/conda.py", line 350, in create

```

Any suggestions would be very welcome - thanks.

**Comment by cbergman** [comment]
Hi @s-andrews 

We have completed testing our fixes for the Relocate2 installation error we identified. Can you please try re-installing the entire McClintock 2 system following the instructions here: https://github.com/bergmanlab/mcclintock#-installing-mcclintock? We have not made any changes to the McClintock conda environment, so in principle you should be able to do a clean install using the updated code as follows:
```
cd mcclintock
git pull
conda activate mcclintock
python3 mcclintock.py --install
```
Please let us know if this solves your installation problem for Relocate2.

Many thanks,
Casey

**Comment by cbergman** [comment]
Hi @s-andrews 

Could you let us know if this issue is resolved on your side? Otherwise, we'll assume it is solved and close at the end of the month.

Best regards,
Casey

**Comment by s-andrews** [comment]
The fix for the relocate package worked and we were able to get that to run.  I still couldn't get the entire test set to work as there was a separate issue with the TEMP2 program where it would just hang indefinitely, but if I excluded that then I could get the rest to run.  I never went back to debug that as we'd got what we needed from our data by that stage.

I've got the logs from an isolated failed run of TEMP2 and have put them up at http://ftp1.babraham.ac.uk/ftpusr50/ (for the next 2 weeks) if you want to take a look.

Thanks for your help with this.
